File: csharp/Structural/10-Facade/Subsystem/NotificationService.cs
namespace Facade.Subsystem;

public class NotificationService
{
    public void NotifyHomeowner(string jobId, string slot)
    {
        Console.WriteLine($"  📧 NotificationService: Homeowner notified — Job #{jobId} scheduled for {slot}.");
    }

    public void NotifyHandyman(string jobId, string handyman)
    {
        Console.WriteLine($"  📱 NotificationService: {handyman} notified of Job #{jobId}.");
    }

    public void NotifySupervisor(string jobId)
    {
        Console.WriteLine($"  🔔 NotificationService: Supervisor notified of new Job #{jobId}.");
    }
}

File: typescript/Structural/10-Facade/Subsystem/NotificationService.ts
export class NotificationService {
  notifyHomeowner(jobId: string, slot: string): void {
    console.log(
      `  📧 NotificationService: Homeowner notified — Job #${jobId} scheduled for ${slot}.`,
    );
  }

  notifyHandyman(jobId: string, handyman: string): void {
    console.log(
      `  📱 NotificationService: ${handyman} notified of Job #${jobId}.`,
    );
  }

  notifySupervisor(jobId: string): void {
    console.log(
      `  🔔 NotificationService: Supervisor notified of new Job #${jobId}.`,
    );
  }
}
